Services/UI/OJS.Services.Ui.Data/Implementations/ContestsDataService.cs

```diff
@@ -25,6 +25,11 @@ public ContestsDataService(OjsDbContext db, IDatesService dates)
             : base(db)
             => this.dates = dates;
 
+        public Task<TServiceModel> GetByProblemId<TServiceModel>(int id)
+            => this.DbSet
+                .FirstOrDefaultAsync(c => c.ProblemGroups.Any(pg => pg.Problems.Any(p => p.Id == id)))
+                .Map<TServiceModel>();
```

Comment:
Do this as follows:
```
=> this.DbSet
  .Where(condition)
  .MapCollection<TServiceModel>()
  .FirstOrDefault();
```

This way, we use the power of AutoMapper and do the _mapping_ on the database, not in memory

Reply:
Fixed.

Reply:
Done.